src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -124,6 +124,7 @@ export class MockHttpSocket extends MockSocket {
    */
   public passthrough(): void {
     const socket = this.createConnection()
+    this.address = () => socket.address()
```

Comment:
Technically, this would be safer:

```suggestion
    this.address = socket.address.bind(socket)
```

In case `this` in the `address()` points to the socket instance, which I think it does. Let's preserve the context.

Reply:
I think both ways preserve the context, but I agree that your solution is more explicit.